File: selenium/src/java/com/thoughtworks/selenium/internal/IdOptionSelectStrategy.java
package com.thoughtworks.selenium.internal;

import com.thoughtworks.webdriver.WebElement;

public class IdOptionSelectStrategy extends BaseOptionSelectStrategy {
	protected boolean selectOption(WebElement option, String selectThis) {
		String id = option.getAttribute("id");
		return selectThis.equals(id);
	}
}

File: selenium/src/csharp/WebdriverBackedSelenium/Internal/SeleniumEmulation/OptionSelectStratgies/IdOptionSelectStrategy.cs
using OpenQA.Selenium;

namespace Selenium.Internal.SeleniumEmulation
{
    internal class IdOptionSelectStrategy : BaseOptionSelectStrategy
    {
        protected override bool SelectOption(IWebElement optionElement, string selectThis)
        {
            string id = optionElement.GetAttribute("id");
            return selectThis.Equals(id);
        }
    }
}